packages/Input/src/Input.js

```diff
@@ -4,25 +4,76 @@ import classNames from "classnames";
 import * as constants from "@paprika/constants/lib/Constants";
 import TimesCircleIcon from "@paprika/icon/lib/TimesCircle";
 import Button from "@paprika/button";
+import useI18n from "@paprika/l10n/lib/useI18n";
 import * as types from "./types";
 import * as sc from "./Input.styles";
 
+function callAll(...fns) {
+  return (...args) => {
+    for (const fn of fns) {
+      if (fn) fn(...args);
+    }
+  };
+}
+
 const Input = React.forwardRef((props, ref) => {
-  const inputClearHandler = e => {
-    e.target.value = "";
-    props.onChange(e);
-    props.onClear();
-    e.stopPropagation();
+  const {
+    a11yText,
+    clearIcon,
+    className,
+    defaultValue,
+    icon,
+    isDisabled,
+    isReadOnly,
+    hasClearButton,
+    hasError,
+    onChange,
+    onClear,
+    size,
+    value,
+    ...moreProps
+  } = props;
+
+  const isControlled = value !== null;
```

Comment:
`null` is a value for any prop you should test for undefined.

React automatically filter those props with the value `undefined` so I think is closer of what we can to compare?

Reply:
The `defaultProps` assign `null` as a default value for `value`.  I can't think of a time I've seen `undefined` as a value in `defaultProps`.